features/call/impl/src/main/kotlin/io/element/android/features/call/impl/utils/ElementCallBaseUrlProvider.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2024 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.call.impl.utils
+
+import com.squareup.anvil.annotations.ContributesBinding
+import io.element.android.features.call.impl.wellknown.CallWellknownAPI
+import io.element.android.libraries.core.coroutine.CoroutineDispatchers
+import io.element.android.libraries.di.AppScope
+import io.element.android.libraries.matrix.api.core.SessionId
+import io.element.android.libraries.network.RetrofitFactory
+import kotlinx.coroutines.withContext
+import retrofit2.HttpException
+import timber.log.Timber
+import java.net.HttpURLConnection
+import javax.inject.Inject
+
+interface ElementCallBaseUrlProvider {
+    suspend fun provides(sessionId: SessionId): String?
+}
+
+@ContributesBinding(AppScope::class)
+class DefaultElementCallBaseUrlProvider @Inject constructor(
+    private val retrofitFactory: RetrofitFactory,
+    private val coroutineDispatchers: CoroutineDispatchers,
+) : ElementCallBaseUrlProvider {
+    override suspend fun provides(sessionId: SessionId): String? = withContext(coroutineDispatchers.io) {
+        val domain = sessionId.value.substringAfter(":")
+        val callWellknownAPI = retrofitFactory.create("https://$domain")
```

Comment:
This is a bit slow to create new retrofit object each time, I'd try to keep a cache if possible